portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -25,11 +27,40 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String text = request.getReader().lines().collect(Collectors.joining(System.lineSeparator()));
-    String name = text.substring(text.indexOf(":")+2, text.lastIndexOf("\""));
-    response.setContentType("text/html;");
-    response.getWriter().println("Hello " + name + "!");
-  }
+    private List<Comment> comments;
+
+    @Override
+    public void init() {
+        comments = new ArrayList<>();
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Comment comment = new Comment(
+            request.getParameter("comment-text"),
+            request.getParameter("author")
+        );
+        comments.add(comment);
+
+        response.sendRedirect("/");
+    }
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String jsonComments = new Gson().toJson(comments);
+
+        response.setContentType("text/html;");
+        response.getWriter().println(jsonComments);
+    }
+}
+
+class Comment {
```

Comment:
I like this a lot!

I'm not suggesting you do this, but have you seen the Builder pattern? It's an interesting tool to have: https://dzone.com/articles/design-patterns-the-builder-pattern

Reply:
I hadn't seen that before! I implemented it to get a feel for it, and I really like it.